### PR TITLE
feat(simple_planning_simulator): add delay converter model

### PR DESCRIPTION
**Pull request by TakumIto**
## Description

Add vehicle delay converter model.

## Tests performed

Work with the planning simulator.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp [hunk 1]
```diff
@@ -257,6 +257,17 @@ void SimplePlanningSimulator::initialize_vehicle_model()
     vehicle_model_ptr_ = std::make_shared<SimModelDelaySteerAccGeared>(
       vel_lim, steer_lim, vel_rate_lim, steer_rate_lim, wheelbase, timer_sampling_time_ms_ / 1000.0,
       acc_time_delay, acc_time_constant, steer_time_delay, steer_time_constant, steer_dead_band);
+  } else if (vehicle_model_type_str == "DELAY_CONVERTER") {
+    vehicle_model_type_ = VehicleModelType::DELAY_CONVERTER;
+    // TODO consider way to give acceleration map path better way
+    const std::string acceleration_map_path = declare_parameter(
+      "acceleration_map_path",
+      std::string("$HOME/projects/pilot-auto.xx1/src/autoware/universe/simulator/"
+                  "simple_planning_simulator/param/acceleration_map.csv"));
```
Comment (kosuke55) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/simple_planning_simulator_core.cpp:266]: default value is not needed

File: simulator/simple_planning_simulator/include/simple_planning_simulator/vehicle_model/sim_model_with_converter.hpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+using namespace raw_vehicle_cmd_converter;
+
+class AccelerationMap
+{
+public:
+  bool readAccelerationMapFromCSV(const std::string & csv_path)
+  {
+    CSVLoader csv(csv_path);
+    std::vector<std::vector<std::string>> table;
+    std::cerr << __LINE__ << std::endl;
```
Comment (kyoichi-sugahara) [inline simulator/simple_planning_simulator/include/simple_planning_simulator/vehicle_model/sim_model_with_converter.hpp:40]: forget to remove?

File: simulator/simple_planning_simulator/include/simple_planning_simulator/vehicle_model/sim_model_with_converter.hpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    for (const auto & acc_vec : acceleration_map_) {
+      interpolated_acc_vec.push_back(interpolation::lerp(vel_index_, acc_vec, clamped_vel));
+    }
+    // calculate throttle
+    // When the desired acceleration is smaller than the throttle area, return min acc
+    // When the desired acceleration is greater than the throttle area, return max acc
+    const double clamped_acc = CSVLoader::clampValue(acc_des, acc_index_, "acceleration: acc");
+    acc = interpolation::lerp(acc_index_, interpolated_acc_vec, clamped_acc);
+
+    return true;
```
Comment (kosuke55) [inline simulator/simple_planning_simulator/include/simple_planning_simulator/vehicle_model/sim_model_with_converter.hpp:66]: this function always return true, so better to return acc as double.

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_with_converter.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  size_t steer_input_queue_size = static_cast<size_t>(round(steer_delay_ / dt));
+  steer_input_queue_.resize(steer_input_queue_size);
+  std::fill(steer_input_queue_.begin(), steer_input_queue_.end(), 0.0);
+}
+
+Eigen::VectorXd SimModelWithConverter::calcModel(
+  const Eigen::VectorXd & state, const Eigen::VectorXd & input)
+{
+  auto sat = [](float64_t val, float64_t u, float64_t l) { return std::max(std::min(val, u), l); };
```
Comment (kosuke55) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_with_converter.cpp:91]: std::clamp ?

**Pull request closed without merging**